pkg/console/subresource/deployment/downloads.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	gracePeriod := int64(1)
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      api.OpenShiftDownloadsName,
+			Namespace: api.OpenShiftConsoleName,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   api.OpenShiftDownloadsName,
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector: nodeSelector(),
+					Tolerations:  tolerations(),
+					Containers: []corev1.Container{
+						downloadsContainer(ipVMode),
+					},
+					TerminationGracePeriodSeconds: &gracePeriod,
+				},
+			},
+		},
+	}
+	return deployment
+}
+
+func nodeSelector() map[string]string {
+	return map[string]string{
+		"kubernetes.io/os": "linux",
+	}
+}
+
+func downloadsContainer(ipVMode config.IPMode) corev1.Container {
+	return corev1.Container{
+		Name:                     api.OpenShiftDownloadsServerName,
+		TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+		Image:                    downloadsImage(),
+		ImagePullPolicy:          corev1.PullIfNotPresent,
+		Ports:                    downloadsPorts(),
+		LivenessProbe:            downloadsLiveness(),
+		ReadinessProbe:           downloadsReadiness(),
+		Command: []string{
+			"/bin/sh",
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: map[corev1.ResourceName]resource.Quantity{
+				corev1.ResourceCPU:    resource.MustParse("10m"),
+				corev1.ResourceMemory: resource.MustParse("50Mi"),
+			},
+		},
+		Args: downloadsArgs(ipVMode),
+	}
+}
+
+// TODO: does this already in manifests/image-references
+// do we need to do anything here?  read it from command line
+// like console deployment IMAGE=<image>?
+func downloadsImage() string {
+	return "registry.svc.ci.openshift.org/openshift:cli-artifacts"
+}
+
+func downloadsPorts() []corev1.ContainerPort {
+	return []corev1.ContainerPort{{
+		Name:          "http",
+		Protocol:      corev1.ProtocolTCP,
+		ContainerPort: 8080,
+	}}
+}
+
+func downloadsLiveness() *corev1.Probe {
+	return &corev1.Probe{
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path:   "/",
+				Port:   intstr.FromInt(8080),
+				Scheme: corev1.URISchemeHTTP,
+			},
+		},
+	}
+}
+
+func downloadsReadiness() *corev1.Probe {
+	probe := downloadsLiveness()
+	probe.FailureThreshold = 3
+	return probe
+}
+
+func downloadsArgs(ipVMode config.IPMode) []string {
+	return []string{
+		"-c",
+		downloadsInlineScript(ipVMode),
+	}
+}
+
+func downloadsInlineScript(ipVMode config.IPMode) string {
+	// defaults to IPv4
+	addressFamily := "AF_INET"
+
+	// if we get a IPv4v6 value, we will have to panic as we don't know how to support both yet
+	if ipVMode == config.IPv4v6Mode {
+		panic(fmt.Sprintf("mode not supported: %v", ipVMode))
+	}
+
+	if ipVMode == config.IPv6Mode {
+		addressFamily = "AF_INET6"
+	}
+
```

Comment:
This isn't working correctly, the `|` is dropped so the pod doesn't start.  Will fix.